packages/consumption/src/modules/requests/itemProcessors/deleteAttribute/DeleteAttributeRequestItemProcessor.ts

```diff
@@ -0,0 +1,121 @@
+        const attribute = await this.consumptionController.attributes.getLocalAttribute(requestItem.attributeId);
+
+        if (!attribute) {
+            return ValidationResult.error(
+                CoreErrors.requests.invalidRequestItem(`The Attribute with the given attributeId '${requestItem.attributeId.toString()}' could not be found.`)
+            );
+        }
+
+        if (!attribute.isOwnSharedAttribute(this.accountController.identity.address)) {
+            return ValidationResult.error(
+                CoreErrors.requests.invalidRequestItem(`The Attribute with the given attributeId '${requestItem.attributeId.toString()}' is not an own shared Attribute.`)
+            );
+        }
+
+        if (attribute.deletionInfo?.deletionStatus === DeletionStatus.DeletedByPeer) {
+            return ValidationResult.error(CoreErrors.requests.invalidRequestItem("The Attribute was already deleted by the peer."));
+        }
+
+        if (attribute.deletionInfo?.deletionStatus === DeletionStatus.ToBeDeletedByPeer) {
+            return ValidationResult.error(CoreErrors.requests.invalidRequestItem("The peer already accepted the deletion of the Attribute."));
+        }
+
+        if (attribute.content.owner.equals(recipient)) {
+            return ValidationResult.error(CoreErrors.requests.invalidRequestItem("The deletion of an own Attribute doesn't need to be requested."));
+        }
+
+        if (!attribute.shareInfo.peer.equals(recipient)) {
+            return ValidationResult.error(
+                CoreErrors.requests.invalidRequestItem("The deletion of a shared Attribute can only be requested from the peer the Attribute is shared with.")
+            );
+        }
+
+        return ValidationResult.success();
+    }
+
+    public override canAccept(_requestItem: DeleteAttributeRequestItem, params: AcceptDeleteAttributeRequestItemParametersJSON, _requestInfo: LocalRequestInfo): ValidationResult {
+        const deletionDate = CoreDate.from(params.deletionDate);
+
+        if (!deletionDate.dateTime.isValid) {
+            return ValidationResult.error(CoreErrors.requests.invalidAcceptParameters("The deletionDate is invalid."));
+        }
+
+        if (deletionDate.isBefore(CoreDate.utc())) {
+            return ValidationResult.error(CoreErrors.requests.invalidAcceptParameters("The deletionDate must be located in the future."));
+        }
+
+        return ValidationResult.success();
+    }
+
+    public override async accept(
+        requestItem: DeleteAttributeRequestItem,
+        params: AcceptDeleteAttributeRequestItemParametersJSON,
+        _requestInfo: LocalRequestInfo
+    ): Promise<DeleteAttributeAcceptResponseItem> {
+        const attribute = await this.consumptionController.attributes.getLocalAttribute(requestItem.attributeId);
+        if (typeof attribute === "undefined") {
+            throw TransportCoreErrors.general.recordNotFound(LocalAttribute, requestItem.attributeId.toString());
```

Comment:
I think this should not be an Error that can be thrown in the `accept` or `canAccept` function because it does not depend on the input of the peer. As long as we don't have the `PermanentErrors` yet, I would be in favour of a more detailed Error message. Something like "may have already been deleted", because this could be a probable reason for the unacceptability of the Request if the peer has already deleted the Attribute (after receiving, but before accepting the Request).

Reply:
Okay, so let's leave it like this for now so that it is consistent and if we have the `PermanentErrors` for the validation later, we could still catch it more precisely beforehand for this specific case if appropriate 👍